OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppManager.swift

```diff
@@ -29,11 +30,21 @@ class InAppManager {
     private let STORED_IN_APP_LIMIT = 50
     private let SYNC_DEBOUNCE_SECONDS = 3600 as TimeInterval
 
+    var finishedInitializationToken: NSObjectProtocol?
+
     // MARK: Initialization
 
-    init(_ config: OptimobileConfig) {
+    init(_ config: OptimobileConfig, httpClient: KSHttpClient) {
+        self.httpClient = httpClient
         presenter = InAppPresenter(displayMode: config.inAppDefaultDisplayMode)
         syncQueue = DispatchQueue(label: "kumulos.in-app.sync")
+
+        finishedInitializationToken = NotificationCenter.default
+            .addObserver(forName: .optimobileInializationFinished, object: nil, queue: nil) { [weak self] notification in
+                guard let self = self else { return }
+                self.sync()
```

Comment:
Is it correct to call `sync()` here and not `syncDebounced()`?

If we directly call `sync()` we will poll the server on every initialisation which is not the previous behaviour and may result in far more requests for the sync endpoint.

LMK if you want to discuss :)

Reply:
why with `handleEnrollmentAndSyncSetup`? -- it does stuff like adding observer and must have been already called during initial partial init . Is it because ordinary sync wouldnt present anything?

syncing here is kinda for in-apps with tickles:
1. kill app
2. receive push
3. open push, app launched ... loading credentials
4. setCredentials ---> now we want to display in-apps that would show if init wasnt delayed

sync debounced with 1h between syncs wouldnt ensure it either

Reply:
The `optimobileInializationFinished` notification triggers the previous regular init flow also for the delayed initialization, simplifying these two cases into one init flow.

Reply:
> Is it because ordinary sync wouldnt present anything?

It will do the intended work.

Reply:
> why with `handleEnrollmentAndSyncSetup`? -- it does stuff like adding observer and must have been already called during initial partial init . Is it because ordinary sync wouldnt present anything?
> 
> syncing here is kinda for in-apps with tickles:
> 
> 1. kill app
> 2. receive push
> 3. open push, app launched ... loading credentials
> 4. setCredentials ---> now we want to display in-apps that would show if init wasnt delayed
> 
> sync debounced with 1h between syncs wouldnt ensure it either

From my understanding of the existing code, it is enough to initialize only the database. 
The things such as adding an observer on `appBecameActive` can be moved to the final init stage, as it is syncing with the server. No need to sync with server without credentials.

Reply:
Thanks Eli, I think `handleEnrollmentAndSyncSetup()` is indeed equivalent w.r.t. what the SDK previously did during init, so seems the correct choice here.

@vvoicehovics interesting question regarding inbox, at first I'm not sure it would matter, but perhaps we should discuss. Practically you can't see inbox items until they're synced anyway, and consent is cached, so this only affects first time init as best I can tell.

Regarding flushing of events, I don't believe this is different (at least on iOS) to how it was before, we always attempted a flush on SDK init here, so I'm less concerned about that.

Let's schedule some time to take a discussion around these topics and look to move this over the line :)

Reply:
(if we find inconsistencies as we go, we should capture under a tech feature perhaps to review)

Reply:
oh, okay, i thought they flushed only when goes bg

you could have inbox items already in db, but consent lost in some way (in any case can clear storage not clearing db?). But y, consent is not used by inbox / unlikely